src/StakingPool/OpenNewEpoch.purs

```diff
@@ -0,0 +1,76 @@
+module StakingPool.OpenNewEpoch where
+
+import Contract.Prelude
+
+import Config.Protocol (mapToProtocolData, readProtocolConfig)
+import Contract.Chain (currentTime)
+import Contract.Log (logInfo')
+import Contract.Monad (Contract, runContract)
+import Contract.ScriptLookups as Lookups
+import Contract.Time (POSIXTime)
+import Contract.TxConstraints as Constraints
+import Ctl.Internal.Types.Interval (from)
+import Effect.Aff (launchAff_)
+import Effect.Exception (throw)
+import Ext.Contract.Time (posixToTimeStamp, roundToSecond)
+import Protocol.Models (Protocol)
+import Protocol.ProtocolScriptInfo (ProtocolScriptInfo(..), getProtocolManagerPkh, getProtocolScriptInfo)
+import Protocol.UserData (ProtocolData, dataToProtocol)
+import Shared.KeyWalletConfig (testnetKeyWalletConfig)
+import Shared.OwnCredentials (OwnCredentials(..), getOwnCreds)
+import Shared.ScriptInfo (ScriptInfo(..), getStakingPoolScriptInfo)
+import Shared.Tx (completeTx)
+import StakingPool.Constraints as StakingPool
+import StakingPool.Datum (PStakingPoolDatum)
+import StakingPoolInfo.Constraints as StakingPoolInfo
+
+runOpenNewEpochFromCli :: Effect Unit
+runOpenNewEpochFromCli = do
+  protocolConfig <- readProtocolConfig
+  let protocolData = mapToProtocolData protocolConfig
+  launchAff_ $ runContract testnetKeyWalletConfig (openNewEpoch protocolData)
+
+openNewEpoch :: ProtocolData -> Contract Unit
```

Comment:
Think we can rename the contract to `checkEpoch` or smth like that. Because we not just creating new epoch, but checking everytime if we need to create it.
Also we can make `checkEpochCompleted` function to highlight the case when we need to create a new one, instead of using `openNewEpoch`. This will help resolve naming issue